plugins/modules/ios_evpn_global.py

```diff
@@ -0,0 +1,356 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# Copyright 2023 Red Hat
+# GNU General Public License v3.0+
+# (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+"""
+The module file for ios_evpn_global
+"""
+
+from __future__ import absolute_import, division, print_function
+
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+module: ios_evpn_global
+short_description: Resource module to configure L2VPN EVPN.
+description: This module provides declarative management of L2VPN EVPN on Cisco IOS network
+  devices.
+version_added: 5.0.0
+author: Padmini Priyadarshini Sivaraj (@PadminiSivaraj)
+notes:
+  - Tested against Cisco IOS device with Version 17.13.01 on Cat9k on CML.
+  - This module works with connection C(network_cli).
+    See U(https://docs.ansible.com/ansible/latest/network/user_guide/platform_ios.html)
+options:
+  config:
+    description: A dictionary of L2VPN Ethernet Virtual Private Network (EVPN) configuration
+    type: dict
+    suboptions:
+      default_gateway:
+        description: Default gateway parameters
+        type: dict
+        suboptions:
+          advertise:
+            description: Advertise Default Gateway MAC/IP routes
+            type: bool
+      flooding_suppression:
+        description: Suppress flooding of broadcast, multicast, and/or unknown unicast packets
+        type: dict
+        suboptions:
+          address_resolution:
+            description: Suppress flooding of Address Resolution and Neighbor Discovery Protocol packets
+            type: dict
+            suboptions:
+              disable:
+                description: Disable flooding suppression
+                type: bool
+      ip:
+        description: IP parameters
+        type: dict
+        suboptions:
+          local_learning:
+            description: IP local learning
+            type: dict
+            suboptions:
+              disable:
+                description: Disable IP local learning
+                type: bool
+      replication_type:
+        description: Method for replicating BUM traffic
+        type: str
+        choices:
+        - ingress
+        - static
+      route_target:
+        description: Route Target VPN Extended Communities
+        type: dict
+        suboptions:
+          auto:
+            description: Automatically set a route-target
+            type: dict
+            suboptions:
+              vni:
+                description: Set vni-based route-target
+                type: bool
+      router_id:
+        description: EVPN router ID
+        type: str
+  running_config:
+    description:
+      - This option is used only with state I(parsed).
+      - The value of this option should be the output received from the IOS
+        device by executing the command B(sh running-config nve | section ^l2vpn evpn$).
```

Comment:
considering this being the scope of the module
```
cat9k1(config)#l2vpn evpn ?
  ethernet-segment  Ethernet segment
  instance          EVPN instance (EVI)
  logging           Configure logging flags
  <cr>              <cr>
```
What about the pending attributes like ethernet-segment / instance/ logging? Should the module consider all of these too?

Reply:
We are treating each of them separately, for instance 'l2vpn evpn instance' is a seperate plugin called 'ios_evpn_evi'.